changes/views.py

```diff
@@ -72,3 +76,62 @@ def post(self, request, *args, **kwargs):
                               action='momconnect_nonloss_optout',
                               data={'reason': 'sms_failure'})
         return Response(status=status.HTTP_201_CREATED)
+
+
+class ReceiveAdminOptout(generics.GenericAPIView):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+
+        admin_serializer = AdminOptoutSerializer(data=request.data)
+        if admin_serializer.is_valid():
+            identity_id = admin_serializer.validated_data["registrant_id"]
+        else:
+            return Response(admin_serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        sbm_client = StageBasedMessagingApiClient(
+            api_url=settings.STAGE_BASED_MESSAGING_URL,
+            auth_token=settings.STAGE_BASED_MESSAGING_TOKEN
+        )
+
+        active_subs = sbm_client.get_subscriptions(
+            {'identity': identity_id, 'active': True}
+        )["results"]
+
+        actions = set()
+        for sub in active_subs:
+            messageset = sbm_client.get_messageset(sub["messageset"])
+            if "nurseconnect" in messageset["short_name"]:
+                actions.add("nurse_optout")
+            elif "pmtct" in messageset["short_name"]:
+                actions.add("pmtct_nonloss_optout")
+            else:
+                actions.add("momconnect_nonloss_optout")
```

Comment:
I'm not sure that we should consider all other messagesets as momconnect. I think we should just ignore any other messagesets that we don't have an optout type for.